### PR TITLE
Added rate limiting to pod deleting

### LINKED ISSUE #6228: When API server becomes unresponsive and comes back after some time NodeController will try to kill all Pods everywhere.
Discussion started in #6193

### PR DESCRIPTION
**Pull request by piosz**
Fixes #6228

### CODE REVIEW

**Comment by piosz** [comment]
RFC @davidopp @gmarek @ddysher

File: cmd/kube-controller-manager/app/controllermanager.go [hunk 1]
```diff
@@ -177,7 +177,7 @@ func (s *CMServer) Run(_ []string) error {
 	}
 
 	nodeController := nodeControllerPkg.NewNodeController(cloud, s.MinionRegexp, s.MachineList, nodeResources,
-		kubeClient, kubeletClient, s.RegisterRetryCount, s.PodEvictionTimeout)
+		kubeClient, kubeletClient, s.RegisterRetryCount, s.PodEvictionTimeout, util.NewTokenBucketRateLimiter(0.01, 10))
```
Comment (brendandburns) [inline cmd/kube-controller-manager/app/controllermanager.go]: Can you make them flags here too?

**Comment by brendandburns** [comment]
LGTM, modulo the flags comment.  If you get to it tonight, that's great.  Otherwise, I'll merge later today and we can handle the flags as a follow on PR.

**Comment by piosz** [comment]
I'll do in 3 hours.

File: cmd/kube-controller-manager/app/controllermanager.go (hunk 1 above)
Comment (piosz) [inline cmd/kube-controller-manager/app/controllermanager.go]: Done

**Comment by davidopp** [comment]
Can you please wait to merge until I've taken a look? I'll do it today.

**Comment by piosz** [comment]
Could you please wait for David?

**Comment by brendandburns** [comment]
yes, will wait.

--brendan

File: cmd/kube-controller-manager/app/controllermanager.go [hunk 2]
```diff
@@ -104,6 +106,8 @@ func (s *CMServer) AddFlags(fs *pflag.FlagSet) {
 	fs.DurationVar(&s.ResourceQuotaSyncPeriod, "resource_quota_sync_period", s.ResourceQuotaSyncPeriod, "The period for syncing quota usage status in the system")
 	fs.DurationVar(&s.NamespaceSyncPeriod, "namespace_sync_period", s.NamespaceSyncPeriod, "The period for syncing namespace life-cycle updates")
 	fs.DurationVar(&s.PodEvictionTimeout, "pod_eviction_timeout", s.PodEvictionTimeout, "The grace peroid for deleting pods on failed nodes.")
+	fs.Float32Var(&s.DeletingPodsQps, "deleting_pods_qps", 0.1, "Number of nodes per second on which pods are deleted in case of node failure.")
+	fs.IntVar(&s.DeletingPodsBurst, "deleting_pods_burst", 10, "Number of nodes on which pods are bursty deleted in case of node failure.")
```
Comment (davidopp) [inline cmd/kube-controller-manager/app/controllermanager.go]: Maybe define what "bursty deleted" means.

**Comment by davidopp** [comment]
LGTM

File: cmd/kube-controller-manager/app/controllermanager.go (hunk 2 above)
Comment (piosz) [inline cmd/kube-controller-manager/app/controllermanager.go]: Added note pointing to RateLimiter.

**Comment by piosz** [comment]
Rebased.

**Comment by gmarek** [comment]
Travis failure is a known race for which fix required this PR.

Thanks, merging.

**Pull request merged**